samcli/lib/samlib/resource_metadata_normalizer.py

```diff
@@ -56,10 +72,59 @@ def _replace_property(property_key, property_value, resource, logical_id):
 
         """
         if property_key and property_value:
-            resource.get(PROPERTIES_KEY, {})[property_key] = property_value
+            nested_keys = property_key.split(".")
```

Comment:
I assume this is to find/update a nested value, like `Code.ImageUri`? If so, can we use `jmespath` instead? We use that lib in `package` to do similar things: https://github.com/aws/aws-sam-cli/blob/ec9f5d89971b823fa01071c96b7de90ea5e5303a/samcli/lib/package/artifact_exporter.py#L94

Reply:
I tried using `set_value_from_jmespath()` but it might not work in this case. We get the template in with `Properties.Code = some_string` and it can't replace a string with a dict. Internally it recurses through the jmespath and sets the next property, so when it gets to the `Properties.Code` string it tries to do `some_string[key] = value` which obviously errors out.

Reply:
I don't understand the explanation. Did you try to use it and it error-ed out?

Reply:
Yes, that's correct.

Reply:
hmm ok. I won't block on this as it's small. I would like to get rid of it to make it easier to maintain but not the end of the world here.